Backend/controllers/userc.js

```diff
@@ -1,21 +1,34 @@
-let user = [];
+const pool = require("../helpers/database");
+const fs = require("fs").promises;
+
+const filePath = "./helpers/user.sql";
+let tableCreated = false;
 
 const createUser = async (req, res) => {
   try {
-    const { email, password } = req.body;
-
-    // Validate email and password
-    if (!email || !password) {
+    const { Username, Password, UserType } = req.body;
+    if (!Username || !Password) {
       return res.status(400).json({ error: "Email and password are required" });
     }
+   
 
-    // Check if email already exists
-    if (user.includes(email)) {
-      return res.status(400).json({ error: "Email already exists" });
+  
+    if (!tableCreated) {
+      const createTableQuery = await fs.readFile(filePath, "utf-8");
+      await pool.query(createTableQuery);
+      tableCreated = true;
     }
 
-    // Add user
-    user.push(email);
+    const insertQuery =
+      "INSERT INTO users (Username, Password, UserType) VALUES (?, ?, ?)";
```

Comment:
What about `UserId`? Firebase Auth will return a unique UserId for each logged in user, that must be stored in the database.

Reply:
Yes that should be here , i will add .